src/parser/mod.rs

```diff
@@ -3615,13 +3615,15 @@ impl<'a> Parser<'a> {
         if has_table {
             let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
             let table_name = self.parse_object_name(false)?;
-            if self.peek_token().token == Token::EOF {
-                Ok(Statement::UNCache {
+            match self.peek_token() {
+                TokenWithLocation {
+                    token: Token::EOF | Token::SemiColon,
+                    ..
```

Comment:
hmm I'm wondering if we don't do an explicitly check for a terminal token after all? looking at the other stmt parse functions seem that's how they handle it, by only parsing the components they need and exiting

Reply:
Yeah that's true, but for some reason this parse function only accepts EOF as and of statement. I'll see if we can remove this check.